cypress/integration/Train/TagAndFrog.spec.js

```diff
@@ -8,72 +8,92 @@ import * as modelPage from '../../support/components/ModelPage'
-    train.SelectAction('Hi')
-
-    train.AbandonDialog()
-
-    cy.WaitForTrainingStatusCompleted()
-    train.CreateNewTrainDialog()
-
-    train.TypeYourMessage('This is Tag.')
-    editDialogModal.TypeAlternativeInput('This is Frog and Tag.')
-    editDialogModal.TypeAlternativeInput('This is Tag and Frog.')
-
-    editDialogModal.VerifyEntityLabelWithinSpecificInput(textEntityPairs[0], 0)
-    editDialogModal.VerifyEntityLabelWithinSpecificInput(textEntityPairs, 1)
-    editDialogModal.VerifyEntityLabelWithinSpecificInput(textEntityPairs, 2)
-
-    editDialogModal.RemoveEntityLabel('Tag', 'multi', 1)
-    editDialogModal.RemoveEntityLabel('Frog', 'multi', 2)
-
-    editDialogModal.ClickScoreActionsButton()
-    editDialogModal.VerifyEntityLabeledDifferentPopupAndClose(textEntityPairs)
-    editDialogModal.ClickScoreActionsButton()
-    editDialogModal.VerifyEntityLabeledDifferentPopupAndAccept(textEntityPairs)
-
-    editDialogModal.VerifyEntityLabeledDifferentPopupAndClose(textEntityPairs)
-    editDialogModal.ClickScoreActionsButton()
-    editDialogModal.VerifyEntityLabeledDifferentPopupAndAccept(textEntityPairs)
-    train.SelectAction('Hi')
+  context('Train - Standard Input', () => {
+    it('Should get an error message after removing single entity label & prevent scoring actions till fixed', () => {
+      train.TypeYourMessage('This is Tag.')
+      editDialogModal.RemoveEntityLabel('Tag', 'multi')
+      editDialogModal.ClickScoreActionsButton()
+      editDialogModal.VerifyEntityLabeledDifferentPopupAndClose([textEntityPairs[0]])
+      editDialogModal.ClickScoreActionsButton()
+      editDialogModal.VerifyEntityLabeledDifferentPopupAndAccept([textEntityPairs[0]])
+      train.SelectAction('Hello')
+    })
+
+    it('Should get an error message after removing a different single entity label & prevent scoring actions till fixed', () => {
+      train.TypeYourMessage('This is Frog and Tag.')
+      editDialogModal.RemoveEntityLabel('Frog', 'multi')
+      editDialogModal.ClickScoreActionsButton()
+      editDialogModal.VerifyEntityLabeledDifferentPopupAndClose(textEntityPairs)
+      editDialogModal.ClickScoreActionsButton()
+      editDialogModal.VerifyEntityLabeledDifferentPopupAndAccept(textEntityPairs)
+      train.SelectAction('Hi')
+    })
+
+    it('Should get an error message after removing two entity labels & prevent scoring actions till fixed', () => {
+      train.TypeYourMessage('This is Tag and Frog.')
+      editDialogModal.RemoveEntityLabel('Tag', 'multi')
+      editDialogModal.RemoveEntityLabel('Frog', 'multi')
+      editDialogModal.ClickScoreActionsButton()
+      editDialogModal.VerifyEntityLabeledDifferentPopupAndClose(textEntityPairs)
+      editDialogModal.ClickScoreActionsButton()
+      editDialogModal.VerifyEntityLabeledDifferentPopupAndAccept(textEntityPairs)
+      train.SelectAction('Hi')
+    })
+  })
 
-    train.Save()
+  context('Setup for next phase', () => {
```

Comment:
Similar to above. Given title says "Setup" I would expect this isn't an actual test and could be included the previous test `after` or next `before` sections